lib/trino-orc/src/main/java/io/trino/orc/reader/ColumnReaders.java

```diff
@@ -47,6 +50,10 @@ public static ColumnReader createColumnReader(
             }
             return new TimeColumnReader(type, column, memoryContext.newLocalMemoryContext(ColumnReaders.class.getSimpleName()));
         }
+        if (type instanceof UuidType) {
```

Comment:
This is actually about Iceberg UUIDs, since UUID is not a native ORC type. We shouldn't be treating this like it's a generic ORC type.  This should validate that `iceberg.binary-type`=`UUID` per the [Iceberg specification for ORC](https://iceberg.apache.org/spec/#orc).

Reply:
This works for Icebrg but I'm not sure it has to be tied to iceberg specifically. This is a very natural way of storing UUIDs so other formats may use it as well.
The validation is already performed in `io.trino.plugin.iceberg.IcebergPageSourceProvider#createOrcPageSource`. Putting it here is problematic as it makes a kind of circular dependency on iceberg and it would complicate testing infra in `OrcTester` a lot.

That said, If you consider it crucial to be here I can prepare changes.

Reply:
We should be conservative about where we use the encoding for UUIDs. If it's only specified by Iceberg, it should be limited to that case. Other systems may end up adopting the same format and we can add it later for them. However, if we do it prematurely, we may end up in a situation where we have to break backward compatibility if they decide to go a different way.

Reply:
I added the `iceberg.binary-type` validation here.
This includes changes to the `OrcType` to support testing the new `UuidColumnReader`.
If anyone sees a better way to do it, let me know